optimum/neuron/modeling_decoder.py

```diff
@@ -49,6 +49,28 @@ def get_exporter(config, task):
     return TasksManager.get_exporter_config_constructor(model_type=config.model_type, exporter="neuron", task=task)()
 
 
+@requires_transformers_neuronx
+def get_available_cores() -> int:
+    """A helper to get the number of available cores.
+
+    This number depends first on the actual number of cores, then on the
+    content of the NEURON_RT_NUM_CORES and NEURON_RT_VISIBLE_CORES variables.
+    """
+    max_cores = len(os.listdir("/sys/class/neuron_device/")) * 2
+    num_cores = os.environ.get("NEURON_RT_NUM_CORES", max_cores)
+    if num_cores != max_cores:
+        num_cores = int(num_cores)
```

Comment:
```suggestion
    num_cores = int(os.environ.get("NEURON_RT_NUM_CORES", max_cores))
```

Reply:
If `NEURON_RT_NUM_CORES` is higher than the max cores, would there be any informative warning or error?

Reply:
You're right: I've added some safeguards.